packages/core/admin/shared/contracts/audit-logs.ts

```diff
@@ -0,0 +1,46 @@
+import { errors } from '@strapi/utils';
+import { Entity, Pagination, SanitizedAdminUser } from './shared';
+
+interface AuditLog extends Pick<Entity, 'id'> {
+  date: string;
+  action: string;
+  /**
+   * TODO: could this be better typed – working on the server-side code could indicate this.
+   * However, we know it's JSON.
+   */
+  payload: Record<string, unknown>;
+  user?: SanitizedAdminUser;
+}
+
+namespace GetAll {
+  export interface Request {
+    body: {};
+    query: {};
+  }
+
+  export interface Response {
+    data: {
+      pagination: Pagination;
+      results: AuditLog[];
+    };
+    error?: errors.ApplicationError;
+  }
+}
+
+namespace Get {
+  export interface Request {
+    body: {};
+    query: {};
+  }
+
+  export interface Params {
+    id: Entity['id'];
+  }
+
+  export interface Response {
+    data: AuditLog;
+    error?: errors.ApplicationError;
+  }
```

Comment:
I hadn't checked the single response before sorry. There's no data key, it's directly the AuditLog object

Reply:
Really? What happens if it errors?

Reply:
@Marc-Roig any idea how to type this case? Should `Reponse` be a union `AuditLog | errors.ApplicationError`?

Reply:
yep, I think this also interests @jhoward1994 , you find a similar case and we were asking about the same thing. We could do

export type Response = AudigLog | { error: errors.ApplicationError }